### PR TITLE
NVHPC: Add headers property to nvhpc

### LINKED ISSUE #32146: Installation issue: py-numpy with nvhpc blas/lapack
### Steps to reproduce the issue

```console
$ spack install py-numpy ^nvhpc
```


### Error message

<details><summary>Error message</summary><pre>
/home/local/KHQ/ryan.krattiger/repos/spack/var/spack/repos/builtin/packages/py-numpy/package.py:223, in set_blas_lapack:                                                                                           
        220        blas_headers = HeaderList([])                                                                                                                                                                   
        221        if "+blas" in spec:                                                                                                                                                                             
        222            blas_libs = spec["blas"].libs                                                                                                                                                               
  >>    223            blas_headers = spec["blas"].headers                                                                                                                                                         
        224                                                                                                                                                                                                        
        225        lapack_libs = LibraryList([])                                                                                                                                                                   
        226        lapack_headers = HeaderList([])                                                                                                                                                                 
                                                                                                                                                                                                                   
See build log for details: 
</pre></details>


### Information on your system

* **Spack:** 0.19.0.dev0 (a68ca0ead9863e6006bc34fd5b2d0e3c205d3109)
* **Python:** 3.8.10
* **Platform:** linux-ubuntu20.04-zen2
* **Concretizer:** clingo


### Additional information

n/a

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

### PR DESCRIPTION
**Pull request by kwryankrattiger**
Fixes spack/spack#32146

### CODE REVIEW

**Comment by haampie** [comment]
Shouldn't these be exposed only when +blas +lapack

File: var/spack/repos/builtin/packages/nvhpc/package.py [hunk 1]
```diff
@@ -386,6 +387,24 @@ def setup_dependent_package(self, module, dependent_spec):
...
+
+        if "+ilp64" in self.spec:
+            prefix = prefix.include.ilp64
+        else:
+            prefix = prefix.include.lp64
+
+        return find_headers(headers, prefix)
+
     @property
     def libs(self):
```
Comment (kwryankrattiger) [inline var/spack/repos/builtin/packages/nvhpc/package.py:409]: This still needs to fix the libraries to support forcing ilp64 and also needs the ability to detect the system default for the headers.

Once that is done, there needs to be a fix to set up the build/runtime environments correctly.

**Comment by kwryankrattiger** [comment]
I think this is more or less fixed in the latest. Closing this iteration.

**Comment by leepbioinfo** [comment]
It still doesn' t work!

**Comment by JGaed** [comment]
Still facing the problem, see https://github.com/spack/spack-packages/issues/953

**Pull request closed without merging**